ksql-serde/src/main/java/io/confluent/ksql/serde/json/KsqlJsonDeserializer.java

```diff
@@ -60,6 +62,10 @@ public KsqlJsonDeserializer(
     if (schema.type() != Type.STRUCT) {
       throw new IllegalArgumentException("KSQL expects all top level schemas to be STRUCTs");
     }
+
+    this.ambiguousMapSchema = schema.fields().size() == 1
+        && schema.fields().get(0).schema().type() == Type.MAP
+        && schema.fields().get(0).schema().keySchema().type() == Type.STRING;
```

Comment:
why have this check? seems like something that would be easy to miss in the future if we support non-string types for map keys. for example: `MAP<INT, INT>` serialized as `{"123": 456}`

Reply:
Because if the key type is not a string then its not ambiguous.  The ambiguity arises when the key is a string and has the same value as the field name.

Reply:
> Numeric field names are not valid
Good point :)